core/src/block.rs

```diff
@@ -19,894 +19,614 @@ use iroha_data_model::{block::*, events::prelude::*, transaction::prelude::*};
+                self.hash()
+            ))
     }
 
-    impl ValidBlock {
-        /// Calculate hash of the current block.
+    /// Create dummy [`ValidBlock`]. Used in tests
+    ///
+    /// # Panics
+    /// If generating keys or block signing fails.
+    #[allow(clippy::restriction)]
+    #[cfg(test)]
+    pub fn new_dummy() -> Self {
+        #[allow(clippy::expect_used)]
+        let timestamp = crate::current_time().as_millis();
+
+        let mut header = BlockHeader {
+            timestamp,
+            consensus_estimation: DEFAULT_CONSENSUS_ESTIMATION_MS,
+            height: 1,
+            view_change_index: 0,
+            previous_block_hash: None,
+            transactions_hash: None,
+            rejected_transactions_hash: None,
+            committed_with_topology: Vec::new(),
+        };
 
-        #[inline]
-        pub fn hash(&self) -> HashOf<Self> {
-            HashOf::new(&self.header).transmute()
-        }
+        let key_pair = KeyPair::generate().unwrap();
+        let signature = SignatureOf::from_hash(key_pair, &HashOf::new(&header).transmute())
+            .expect("Signing of new block failed.");
+        let signatures = SignaturesOf::from(signature);
 
-        /// Sign this block and get [`SignedBlock`].
-        ///
-        /// # Errors
-        /// Fails if signature generation fails
-        pub fn sign(self, key_pair: KeyPair) -> Result<SignedBlock> {
-            let signature = SignatureOf::from_hash(key_pair, &self.hash().transmute())
-                .wrap_err(format!("Failed to sign block with hash {}", self.hash()))?;
-            let signatures = SignaturesOf::from(signature);
-            Ok(SignedBlock {
-                header: self.header,
-                rejected_transactions: self.rejected_transactions,
-                transactions: self.transactions,
-                event_recommendations: self.event_recommendations,
-                signatures,
-            })
+        Self {
+            header,
+            rejected_transactions: Vec::new(),
+            transactions: Vec::new(),
+            event_recommendations: Vec::new(),
+            signatures,
         }
     }
 }
 
-/// Signed block related structures and implementations.
-mod signed {
-    use super::*;
-
-    /// After receiving first signature, [`ValidBlock`] can transform into [`SignedBlock`].
-    #[derive(Debug, Clone)]
-    pub struct SignedBlock {
-        /// Block header
-        pub header: BlockHeader,
-        /// Array of all rejected transactions in this block.
-        pub rejected_transactions: Vec<VersionedRejectedTransaction>,
-        /// Array of all valid transactions in this block.
-        pub transactions: Vec<VersionedValidTransaction>,
-        /// Signatures of peers which approved this block.
-        pub signatures: SignaturesOf<Self>,
-        /// Event recommendations.
-        pub event_recommendations: Vec<Event>,
-    }
+pub trait Auditable: Sized {
```

Comment:
Nice to see traits.

Reply:
```suggestion
pub trait Revalidate: Sized {
```
Rust has a different naming convention for interfaces.